vars/testsPublishResults.groovy

```diff
@@ -175,6 +181,21 @@ def publishCucumberReport(Map settings = [:]) {
     }
 }
 
+def publishHtmlReport(Map settings = [:]) {
+    if (settings.active) {
+        def pattern = settings.get('pattern')
+
+        publishHTML(target: [
+            allowMissing         : settings.get('allowMissing'),
+            alwaysLinkToLastBuild: settings.get('alwaysLinkToLastBuild'),
+            keepAll              : settings.get('keepAll'),
+            reportDir            : settings.get('reportDir'),
+            reportFiles          : pattern,
```

Comment:
```suggestion
        publishHTML(target: [
            allowMissing         : settings.get('allowMissing'),
            alwaysLinkToLastBuild: settings.get('alwaysLinkToLastBuild'),
            keepAll              : settings.get('keepAll'),
            reportDir            : settings.get('reportDir'),
            reportFiles          : settings.get('pattern'),
```

Reply:
Yeah I was also thinking about it, but followed what is used throughout the step already. The config parameter `pattern` is always first saved to a variable. I guess this is due to the parameter of the plugins usually being named differently but meaning the same thing. As in this case where I use it for the plugin parameter `reportFiles`.
What would be your take on this?

Reply:
I noticed this before. In some steps, the pattern is used for archiving after the step invocation. So there it makes sense. For other steps, it could be that it was copied just like you did.

Reply:
Ah good point, just noticed that `pattern` is also used for the `archiveResults` call.

Reply:
Then I think you are right with your suggestion!